specs/_features/eip7251/beacon-chain.md

```diff
@@ -590,10 +590,19 @@ def process_pending_balance_deposits(state: BeaconState) -> None:
     next_deposit_index = 0
 
     for deposit in state.pending_balance_deposits:
-        if processed_amount + deposit.amount > available_for_processing:
+        # Validator is exited, balance never becomes active, no need to consume churn
+        if validator.exit_epoch <= get_current_epoch(state):
+            increase_balance(state, deposit.index, deposit.amount)
+        # Validator is active but exiting, postpone the deposit
+        elif validator.exit_epoch < FAR_FUTURE_EPOCH:
+            state.pending_balance_deposits.append(deposit)
```

Comment:
In case when the queue contains top-ups to exiting validators appending them within the loop will cause the loop to never end